packages/beacon-node/src/chain/blocks/index.ts

```diff
@@ -141,6 +141,23 @@ export async function processBlocks(
       }
     }
 
+    // clean db if we don't have blocks in forkchoice but already persisted them to db
+    const removeBlockPromises: Promise<void>[] = [];
+    for (const blockInput of blocks) {
+      const {block, type} = blockInput;
+      const blockRoot = this.config.getForkTypes(block.message.slot).BeaconBlock.hashTreeRoot(block.message);
+      if (!this.forkChoice.hasBlock(blockRoot)) {
+        removeBlockPromises.push(this.db.block.remove(block));
+        if (type === BlockInputType.postDeneb) {
+          const {blobs} = blockInput;
+          removeBlockPromises.push(this.db.blobsSidecar.remove(blobs));
+        }
+      }
+    }
+    Promise.all(removeBlockPromises).catch((e) => {
+      this.logger.verbose("Failed to remove blocks", {slots: blocks.map((b) => b.block.message.slot).join(",")}, e);
+    });
```

Comment:
I would not bundle in a single Promise.all since then separate errors are reported once, just attach a .catch (two total) to each call above